library-backend/library-api/src/main/java/org/internship/library/api/BookRepository.java

```diff
@@ -0,0 +1,39 @@
+package org.internship.library.api;
+
+/**
+ * This interface defines the behaviour of Book repository
+ */
+public interface BookRepository {
+
+    /**
+     * Finds the book by the given id
+     * Returns EntityNotFoundException on the contrary
+     * Returns Bad_Request(400) if Book fields are incomplete
```

Comment:
The repository layer of an application should not know about the API layer.

e.g. if we use this repository for another API, which is not REST, your comment is invalid. Remove the 3rd line.

Reply:
Same for all of the methods.